components/prism-factor.js

```diff
@@ -0,0 +1,370 @@
+				lookbehind: true,
+				alias: 'string',
+				inside: string_inside
+			}
+		],
+
+		'special-using': {
+			pattern: /(^|\s)USING:(?:\s\S+)*(?=\s+;(?:\s|$))/,
+			lookbehind: true,
+			alias: 'function',
+			inside: {
+				// this is essentially a regex for vocab names, which i don't want to specify
+				// but the USING: gets picked up as a vocab name
+				string: {
+					pattern: /(\s)[^:\s]+/,
+					lookbehind: true
+				}
+			}
+		},
+
+		/* this description of stack effect literal syntax is not complete and not as specific as theoretically possible
+			trying to do better is more work and regex-computation-time than it's worth though.
+			- we'd like to have the "delimiter" parts of the stack effect [ (, --, and ) ] be a different (less-important or comment-like) colour to the stack effect contents
+			- we'd like if nested stack effects were treated as such rather than just appearing flat
+			- we'd like if the following variable name conventions were recognised specifically:
+				special row variables = ..a b..
+				type and stack effect annotations end with a colon = ( quot: ( a: ( -- ) -- b ) -- x ), ( x: number -- )
+				word throws unconditional error = *
+				any other word-like variable name = a ? q' etc
+
+			https://docs.factorcode.org/content/article-effects.html
+
+			these are pretty complicated to highlight properly without a real parser, and therefore out of scope
+			the old pattern, which may be later useful, was: (^|\s)(?:call|execute|eval)?\((?:\s+[^"\r\n\t ]\S*)*?\s+--(?:\s+[^"\n\t ]\S*)*?\s+\)(?=\s|$)
+		*/
+
+		// current solution is not great
+		'stack-effect-delimiter': [
+			{
+				// opening parenthesis
+				pattern: /(^|\s)(?:call|execute|eval)?\((?=\s)/,
+				lookbehind: true,
+				alias: 'operator'
+			},
+			{
+				// middle --
+				pattern: /(\s)--(?=\s)/,
+				lookbehind: true,
+				alias: 'operator'
+			},
+			{
+				// closing parenthesis
+				pattern: /(\s)\)(?=\s|$)/,
+				lookbehind: true,
+				alias: 'operator'
+			}
+		],
+
+		'combinators': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'keyword'
+		},
+
+		'kernel-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'sequences-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'math-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'constructor-word': {
+			// <array> but not <=>
+			pattern: /(^|\s)<(?!=+>|-+>)\S+>(?=\s|$)/,
+			lookbehind: true,
+			alias: 'keyword'
+		},
+
+		'other-builtin-syntax': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'operator'
+		},
+
+		'conventionally-named-word': {
+			// for certain word naming conventions like +lt+, sequence?, readers>>, >>setters, writers<<
+			pattern: /(^|\s)(?!")(?:set-\S+|change-\S+|>[^>\s]+|[^:>\s]+>|[^>\s]+>[^>\s]+|\+[^\+\s]+\+|[^?\s]+\?|\?[^?\s]+|[^>\s]+>>|>>[^>\s]+|[^<\s]+<<|\([^()\s]+\)|[^!\s]+!|[^\s]\S*\*|[^\s]\S*\.)(?=\s|$)/,
```

Comment:
`[^\s]` == `\S`, so either make `[^\s]\S*` -> `\S+` or add the suffix to the set (e.g. `[^\s]\S*\*` -> `[^*\s]\S*\*`).
```suggestion
			pattern: /(^|\s)(?!")(?:set-\S+|change-\S+|>[^>\s]+|[^:>\s]+>|[^>\s]+>[^>\s]+|\+[^+\s]+\+|[^?\s]+\?|\?[^?\s]+|[^>\s]+>>|>>[^>\s]+|[^<\s]+<<|\([^()\s]+\)|[^!\s]+!|\S+\*|\S+\.)(?=\s|$)/,
```

Reply:
i missed including the suffix in the set, thanks